scenes/test_menu.txt

```diff
@@ -0,0 +1,173 @@
+*gosub test_assert_equal "[i]<-- Previous[/i]" cslib_choice_2
+*gosub test_assert_equal "6" cslib_choice_3
+*gosub test_assert_equal "7" cslib_choice_4
+*gosub test_assert_equal "8" cslib_choice_5
+*gosub test_assert_equal "9" cslib_choice_6
+*gosub test_assert_equal "10" cslib_choice_7
+*gosub test_assert_equal "[i]Next -->[/i]" cslib_choice_8
+*gosub test_finish
+
+*gosub test_start "simple - multi-page (page 3, option 1)"
+*set cslib_choice_select "2,2,3"
+*gosub_scene cslib_menu build_simple 1 2 3 4 5 6 7 8 9 10 11
+*gosub test_assert_equal 3 cslib_choice_count
+*gosub test_assert_equal 11 cslib_ret
+*gosub test_assert_equal "Page (3 / 3)" cslib_choice_1
+*gosub test_assert_equal "[i]<-- Previous[/i]" cslib_choice_2
+*gosub test_assert_equal "11" cslib_choice_3
+*gosub test_finish
+
+
+*comment :::::: ARRAY tests ::::::
+*comment array - init test data
+*set test_menu_opt_count 7
+*set test_menu_opt_1 "Odd"
+*set test_menu_opt_visible_1 false
+*set test_menu_opt_2 "Even"
+*set test_menu_opt_visible_2 true
+*set test_menu_opt_3 "Odd"
+*set test_menu_opt_visible_3 false
+*set test_menu_opt_4 "Even"
+*set test_menu_opt_visible_4 true
+*set test_menu_opt_5 "Odd"
+*set test_menu_opt_visible_5 false
+*set test_menu_opt_6 "Even"
+*set test_menu_opt_visible_6 true
+*set test_menu_opt_7 "Odd"
+*set test_menu_opt_visible_7 false
+
+*gosub test_start "array - multi-page (with return, page 1, selecting '4')"
+*set cslib_choice_select "6"
+*gosub_scene cslib_menu build_array "const_num" true
+*gosub test_assert_equal 8 cslib_choice_count
+*gosub test_assert_equal 4 cslib_ret
+*gosub test_assert_equal "Page (1 / 2)" cslib_choice_1
+*gosub test_assert_equal "[i]Next -->[/i]" cslib_choice_2
+*gosub test_assert_equal "1" cslib_choice_3
+*gosub test_assert_equal "2" cslib_choice_4
+*gosub test_assert_equal "3" cslib_choice_5
+*gosub test_assert_equal "4" cslib_choice_6
+*gosub test_assert_equal "5" cslib_choice_7
+*gosub test_assert_equal "[return]" cslib_choice_8
+*gosub test_finish
+
+*gosub test_start "array - multi-page (with return, page 2, selecting 'return')"
+*set cslib_choice_select "2,7"
+*gosub_scene cslib_menu build_array "const_num" true
+*gosub test_assert_equal 7 cslib_choice_count
+*gosub test_assert_equal 0 cslib_ret
+*gosub test_assert_equal "Page (2 / 2)" cslib_choice_1
+*gosub test_assert_equal "[i]<-- Previous[/i]" cslib_choice_2
+*gosub test_assert_equal "6" cslib_choice_3
+*gosub test_assert_equal "7" cslib_choice_4
+*gosub test_assert_equal "8" cslib_choice_5
+*gosub test_assert_equal "9" cslib_choice_6
+*gosub test_assert_equal "[return]" cslib_choice_7
+*gosub test_finish
+
+*comment *gosub test_start "array - filter"
+*comment *set cslib_choice_select "2"
+*comment *gosub_scene cslib_menu build_array "test_menu_opt" false *comment "visible" "MATCH" true
+*comment *gosub test_assert_equal 3 cslib_choice_count
+*comment *gosub test_assert_equal 1 cslib_ret
+*comment *gosub test_assert_equal "Even" cslib_choice_1
```

Comment:
are you not testing filters?

Reply:
Apparently not, that's embarrassing, but probably what you get for pushing code at stupid o'clock.
Thanks for catching, I shall fix.